### PR TITLE
add tests for wandb.sklearn

### PR DESCRIPTION
**Pull request by charlesfrye**
Description
-----------

Adds tests for `wandb.sklearn`. See closed PR https://github.com/wandb/client/pull/2867 for details.

Testing
-------

I ran the tests locally with `yea`.

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------

NO RELEASE NOTES

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: functional_tests/sklearn/plot_clusterer-basic.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+plugin:
+    - wandb
+depend:
+    requirements:
+        - numpy
+        - scikit-learn
+assert:
+    - :wandb:runs_len: 1
+    - :wandb:runs[0][exitcode]: 0
+    - :yea:exit: 0
```
Comment (raubitsj) [inline functional_tests/sklearn/plot_clusterer-basic.py:14]: Please add an assert for the plots.. here is an example for one:
```
assert:
  - :wandb:run[0][summary][elbow_curve][_type]: table-file
  - :wandb:run[0][summary][elbow_curve][ncols]: 3
  - :wandb:run[0][summary][elbow_curve][nrows]: 5
```

similar for silhouette_plot

to figure these out run:
```
yea --debug run myyeafile.py
```

**Comment by BBischof** [comment]
Awesome; thanks for adding these.

File: functional_tests/sklearn/plot_clusterer-basic.py (hunk 1 above)
Comment (charlesfrye) [reply functional_tests/sklearn/plot_clusterer-basic.py:20]: done, thanks jeff!

**Comment by charlesfrye** [comment]
The `plot_classifier` test turned out to break in the `py37` environment ([example](https://app.circleci.com/pipelines/github/wandb/client/8717/workflows/c5d94f18-ac90-438f-bd4b-d6b92cbb4c4b/jobs/87191/steps)), possibly due to the same random seed causing different behavior there than in the environment where I was testing. I wasn't able to directly replicate, but the proximal cause is clear.

Specifically, the train/test split in `plot_calibration_curve` can result in a training set that doesn't have a sufficient number of examples (>=2 for each of 2 classes) for the `model.fit` call to run. This reflects a broader issue: the construction of the training and test sets for the comparison classifiers is based on a [sklearn example](https://scikit-learn.org/stable/auto_examples/calibration/plot_compare_calibration.html), which uses a synthetic dataset of fixed, very large size and has a 999 to 1 test/train split (this was also the cause of the issue fixed in #2791). Our example has a 98 to 2 test/train split. This doesn't really reflect the way training and test splits work in practice, where something like a 2 to 8 ratio (note inversion!) is most common.

This extreme split makes the produced plot hard to interpret -- the classifiers we train will be trained on a very small fraction of the training data provided, so it's unclear whether they can be fairly compared with the user's model, and the small size of the training data for typical causes fluctuations to dominate in the chart.

Correct behavior is tricky and depends on how we want this to be used. You might want the training set for the comparison to be the same as that used to train the provided classifier, and therefore ask the user to provide a test set as well.

For now, I've resolved by changing the train/test split to 9 to 1, instead of 98 to 2, which should avoid this issue on reasonably-sized datasets. This is a small behavior change.

**Comment by charlesfrye** [comment]
@raubitsj this is ready for review!

File: tests/integrations/test_sklearn.py [hunk 2]
```diff
@@ -90,91 +83,6 @@ def dummy_data(request):
     return (data, labels)
 
 
-def test_learning_curve(dummy_classifier):
```
Comment (raubitsj) [inline tests/integrations/test_sklearn.py:93]: Are these tests no good?   Duplicating functional and unittests is a perfectly valid thing to do.

But if the unittests are poor, im fine with removing them, i havent looked carefully enough to make a judgement.  what was the reasoning for removing?

Comment (charlesfrye) [reply tests/integrations/test_sklearn.py:93]: Most were testing at a lower level than we'd like to test -- testing functionality in the midlevel of the API, namely `roc`, `learning_curve`, `confusion_matrix` and `precision_recall`. I suspect they tested this level because it was infeasible, without `yea`, to test the top level of the API, which calls `wandb.log`.

**Review by raubitsj — APPROVED** [review]

File: tests/integrations/test_sklearn.py (hunk 2 above)
Comment (raubitsj) [reply tests/integrations/test_sklearn.py:93]: Thank you for explaining!  looks good
